tests/rados/test_osd_inconsistency_pg.py

```diff
@@ -0,0 +1,110 @@
+"""
+The script is to generate and verification of the inconsistent pg
+"""
+
+import random
+import time
+import traceback
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from tests.rados.rados_test_util import create_pools
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    client_node = ceph_cluster.get_nodes(role="client")[0]
+    pool_obj = PoolFunctions(node=cephadm)
+    try:
+        log.info("Creating the pool replicated pool")
+        create_pools(config, rados_obj, client_node)
+        log.info("Pool creation completed")
+        pool_name = config["create_pools"][0]["create_pool"]["pool_name"]
+
+        # Adding an object
+        oname = "UNIQUEOBJECT{i}".format(i=random.randint(0, 10000))
+        log.info(f"Creating the object-{oname} in the pool-{pool_name}")
+        pool_obj.do_rados_put(client=client_node, pool=pool_name, obj_name=oname)
+        log.info(f"The object-{oname} is created in the pool-{pool_name}")
+        osd_map_output = rados_obj.get_osd_map(pool=pool_name, obj=oname)
+        primary_osd = osd_map_output["acting_primary"]
+        log.info(f"The object stored in the primary osd number-{primary_osd}")
+        pg_id = osd_map_output["pgid"]
+        log.info(f"The object {oname} is created in the pg-{pg_id}")
+        # Adding the key value to object
+        log.info("Adding a key to the object")
+        cmd_setkey = f"rados -p {pool_name} setomapval {oname} key0 value0"
+        client_node.exec_command(cmd=cmd_setkey, sudo=True)
+        # stoping the OSD
+        if not rados_obj.change_osd_state(action="stop", target=primary_osd):
+            log.error(f"Unable to stop the OSD : {primary_osd}")
+            raise Exception("Execution error")
+        acting_osd_node = rados_obj.fetch_host_node(
+            daemon_type="osd", daemon_id=primary_osd
+        )
+        cmd_base = f"cephadm shell --name osd.{primary_osd} --"
+        log.info(f"Corrupting the object-{oname} key.")
+        cmd_get_snaObj = (
+            f"{cmd_base} ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-"
+            f"{primary_osd} --pgid {pg_id} {oname} set-omap key_corrupt1 /etc/hosts"
+        )
+        acting_osd_node.exec_command(sudo=True, cmd=cmd_get_snaObj)
+        if not rados_obj.change_osd_state(action="start", target=primary_osd):
+            log.error(f"Unable to start the OSD : {primary_osd}")
+            raise Exception("Execution error")
+            # Wait until pg active+clean state
+        while True:
```

Comment:
Instead of while loop, I would suggest to use method : wait_for_active_clean_pgs so that for any reason if we dont get he active+clean state we dont end up looping forever. The default timeout for the method is 3600, based on execution logs if needed can be increased if need be.